docs/Hosting_Application_Usage.md

```diff
@@ -0,0 +1,134 @@
+# How to Use ONNX Hosting Application Provide REST API for Prediction
+
+ONNX hosting application is designed to provide REST API for prediction service. The command line is quite simple as below. Only `model_path` is MUST-HAVE to start a server.
+
+```
+~./onnxruntime_hosting --help
+ONNX Hosting: host an ONNX model for inferencing with ONNXRuntime
+Allowed options:
+  -h [ --help ]                   Shows a help message and exits
+  --logging_level arg (=verbose)  Logging level. Allowed options (case 
+                                  sensitive): verbose, info, warning, error, 
+                                  fatal
+  -m [ --model_path ] arg         Path to ONNX model
+  -a [ --address ] arg (=0.0.0.0) The base HTTP address
+  --http_port arg (=8001)         HTTP port to listen to requests
+  --num_http_threads arg (=8)     Number of http threads
+
+```
+
+## Start the Server
+
+To host an ONNX model as REST API server, you simply need to run as:
+
+```
+./onnxruntime_hosting -m /<your>/<model>/<path>
+```
+
+The prediction request is in this format:
+
+```
+http://<your_ip_address>:<port>/v1/models/<your-model-name>/versions/<your-version>:predict
+```
+
+**Note**: Since we currently only support one model, the model name and version could be any string length > 0.
+
+## Request and Response Payload
+
+The HTTP request could be ProtoBuf message in binary format or JSON format. The HTTP request header field `Content-Type` will tell the server how to handle the request payload. It is a MUST-HAVE field in request headers. Requests missing `Content-Type` will be rejected as `400 Bad Request`.
+
+* For `"Content-Type: application/json"`, the payload will be deserialized as JSON string in UTF-8 format
+* For `"Content-Type: application/vnd.google.protobuf"`, `"Content-Type: application/x-protobuf"` or `"Content-Type: application/octet-stream"`, the payload will be consumed as protobuf message directly.
+
+The ProtoBuf definition could be found [here](https://github.com/Microsoft/onnxruntime/blob/master/onnxruntime/hosting/protobuf/predict.proto). You could use it for your preferred programming language.
```

Comment:
```suggestion
The Protobuf definition can be found [here](https://github.com/Microsoft/onnxruntime/blob/master/onnxruntime/hosting/protobuf/predict.proto). You can use it for your preferred programming language.
```

Reply:
Checked [Protocol Buffers GitHub repo](https://github.com/protocolbuffers/protobuf), they either use `Protocol Buffers` or `protobuf`. So I will update the doc with `protobuf` all in lower case.